vendor-extra/JatsContentBundle/tests/EventListener/BuildView/FrontArticleTypeContentMetaListenerTest.php

```diff
@@ -0,0 +1,247 @@
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<front xmlns="http://jats.nlm.nih.gov"/>
+XML
+        );
+
+        $event = new BuildViewEvent($element, new View('@LiberoPatterns/content-meta.html.twig'));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertSame('@LiberoPatterns/content-meta.html.twig', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_there_is_no_article_type() : void
+    {
+        $listener = new FrontArticleTypeContentMetaListener(
+            new IdentityTranslator(),
+            ['research-article' => 'Research article']
+        );
+
+        $element = $this->loadElement(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<article xmlns="http://jats.nlm.nih.gov">
+    <front/>
+</article>
+XML
+        )->firstElementChild;
+
+        $event = new BuildViewEvent($element, new View('@LiberoPatterns/content-meta.html.twig'));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertSame('@LiberoPatterns/content-meta.html.twig', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_there_is_already_a_type_set() : void
+    {
+        $listener = new FrontArticleTypeContentMetaListener(
+            new IdentityTranslator(),
+            ['research-article' => 'Research article']
+        );
+
+        $element = $this->loadElement(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<article article-type="research-article" xmlns="http://jats.nlm.nih.gov">
+    <front/>
+</article>
+XML
+        )->firstElementChild;
+
+        $event = new BuildViewEvent(
+            $element,
+            new View(
+                '@LiberoPatterns/content-meta.html.twig',
+                ['items' => ['type' => 'foo']]
+            )
+        );
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertSame('@LiberoPatterns/content-meta.html.twig', $view->getTemplate());
+        $this->assertSame(['items' => ['type' => 'foo']], $view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_the_type_is_unknown() : void
+    {
+        $listener = new FrontArticleTypeContentMetaListener(
+            new IdentityTranslator(),
+            ['research-article' => 'Research article']
+        );
+
+        $element = $this->loadElement(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<jats:article article-type="not-research-article" xmlns:jats="http://jats.nlm.nih.gov">
+    <jats:front/>
+</jats:article>
+XML
+        )->firstElementChild;
```

Comment:
the negative tests are long-winded but they cover the possible inputs, so I don't see an easy way to condense them